src/NHibernate/Mapping/ForeignKey.cs

```diff
@@ -122,12 +122,14 @@ private void AlignColumns(Table referencedTable)
 				sb.Append("])");
 				throw new FKUnmatchingColumnsException(sb.ToString());
 			}
-			IEnumerator<Column> fkCols = ColumnIterator.GetEnumerator();
-			IEnumerator<Column> pkCols = referencedTable.PrimaryKey.ColumnIterator.GetEnumerator();
 
-			while (fkCols.MoveNext() && pkCols.MoveNext())
+			using (var fkCols = ColumnIterator.GetEnumerator())
+			using (var pkCols = referencedTable.PrimaryKey.ColumnIterator.GetEnumerator())
 			{
-				fkCols.Current.Length = pkCols.Current.Length;
+				while (fkCols.MoveNext() && pkCols.MoveNext())
```

Comment:
Granted, I am not familiar with `Zip`. Using it would be awkward. It does produce a third sequence we do not need here, and we would have to artificially iterate over it while using a result operator having some side effects. Something like:

    fkCols.Zip(pkCols, (fkc, pkc) => fkc.Length = pkc.Length).Count();

That should work but looks quite ugly and counter-intuitive to me (and will probably cause some warnings).